libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -1266,6 +1268,7 @@ int HTTPClient::handleHeaderResponse()
 
             if(headerLine.startsWith("HTTP/1.")) {
                 _returnCode = headerLine.substring(9, headerLine.indexOf(' ', 9)).toInt();
+                _canReuse = (_returnCode != '0');
```

Comment:
You're checking the return code, not the HTTP version to set canReuse.  Shouldn't you just be looking at header[7] and if it's != 0 you can reuse?

Reply:
It would indeed feel less "hacky" and easier to understand.